### PR TITLE
refactor: update querying interface

### PR DESCRIPTION
**Pull request by JabobKrauskopf**
Updates in PR:

- fixed a few errors in interface
- add operand cloning
- removed `any` because it is quite complicated to implement and I want to see if we can live without it
- `NodeIndicesOperand` and `NodeIndexOperand` are now their own thing
- minor fixes

Expected tests to fail (same as in the previous PR):
- `medmodels/medrecord/tests/test_builder.py`
- `medmodels/medrecord/tests/test_datatype.py`
- `medmodels/medrecord/tests/test_indexers.py`
- `medmodels/medrecord/tests/test_medrecord.py`
- `medmodels/medrecord/tests/test_overview.py`
- `medmodels/medrecord/tests/test_schema.py`
- `medmodels/treatment_effect/matching/tests/test_classic_distance_models.py`
- `medmodels/treatment_effect/matching/tests/test_metrics.py`
- `medmodels/treatment_effect/matching/tests/test_propensity_score.py`
- `medmodels/treatment_effect/tests/test_continuous_estimators.py`
- `medmodels/treatment_effect/tests/test_temporal_analysis.py`
- `medmodels/treatment_effect/tests/test_treatment_effect.py`

### CODE REVIEW

File: medmodels/medrecord/querying.pyi [hunk 1]
```diff
@@ -19,314 +20,377 @@ else:
...
-        self, attribute: Union[MedRecordAttribute, List[MedRecordAttribute]]
+        self, attribute: Union[MedRecordAttribute, SingleAttributeOperand]
     ) -> None: ...
     def incoming_edges(self) -> EdgeOperand: ...
     def outgoing_edges(self) -> EdgeOperand: ...
     def neighbors(
         self, edge_direction: EdgeDirection = EdgeDirection.OUTGOING
     ) -> NodeOperand: ...
     def either_or(self, either: NodeQuery, or_: NodeQuery) -> None: ...
+    def clone(self) -> NodeOperand: ...
```
Comment (MarIniOnz) [inline medmodels/medrecord/querying.pyi:69]: What is the purpose of this `clone` operands on Node and Edges? Just curious

File: medmodels/medrecord/querying.pyi [hunk 2]
```diff
@@ -19,314 +20,377 @@ else:
...
+    def last_where(
+        self, query: Callable[[SingleValueOperand], None]
+    ) -> SingleValueOperand: ...
     def mean(self) -> SingleValueOperand: ...
-    def any(self, query: Callable[[SingleValueOperand], None]) -> None: ...
+    def median(self) -> SingleValueOperand: ...
+    def mode(self) -> SingleValueOperand: ...
+    def std(self) -> SingleValueOperand: ...
+    def var(self) -> SingleValueOperand: ...
+    def count(self) -> SingleValueOperand: ...
```
Comment (MarIniOnz) [inline medmodels/medrecord/querying.pyi:100]: Is the idea behind `count` to show the number of available values?

File: medmodels/medrecord/querying.pyi [hunk 3]
```diff
@@ -19,314 +20,377 @@ else:
...
     ) -> None: ...
     def source_node(self) -> NodeOperand: ...
     def target_node(self) -> NodeOperand: ...
     def either_or(self, either: EdgeQuery, or_: EdgeQuery) -> None: ...
+    def clone(self) -> EdgeOperand: ...
 
 class MultipleValuesOperand:
     def max(self) -> SingleValueOperand: ...
     def min(self) -> SingleValueOperand: ...
+    def first_where(
```
Comment (MarIniOnz) [inline medmodels/medrecord/querying.pyi:89]: What is this first or last used for?

**Review by MarIniOnz — APPROVED** [review]
One step closer! Minor explanations needed but not real changes.

File: medmodels/medrecord/querying.pyi (hunk 1 above)
Comment (JabobKrauskopf) [reply medmodels/medrecord/querying.pyi:69]: when you want to save the operand with its current operations and use it later on.

e.g.
```python
def query(node: NodeOperand):
    node.attribute("foo").less_than(5)
    saved_operand = node.clone()
    node.has_attribute("bar")
    ...
    # use `saved_operand` here for other operations where you just want the operand to be filtered with `node.attribute("foo").less_than(5)` and not `has_attribute("bar")`
    ....
```

File: medmodels/medrecord/querying.pyi (hunk 2 above)
Comment (JabobKrauskopf) [reply medmodels/medrecord/querying.pyi:100]: not to show the number of available values (e.g. print or anything) but to use the number of available values in another part of the query, yes

File: medmodels/medrecord/querying.pyi (hunk 1 above)
Comment (MarIniOnz) [reply medmodels/medrecord/querying.pyi:69]: good idea!

**Pull request merged**